### PR TITLE
Get rid of unused variables part3

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: std/array.d [hunk 1]
```diff
@@ -163,7 +163,7 @@ if (isIterable!Range && !isNarrowString!Range && !isInfinite!Range)
     struct Foo
     {
         int a;
-        void opAssign(Foo foo)
+        void opAssign(Foo)
```
Comment (JackStouffer) [inline std/array.d]: what the?

you can have a parameter without a name?

File: std/array.d [hunk 2]
```diff
@@ -189,6 +189,7 @@ if (isIterable!Range && !isNarrowString!Range && !isInfinite!Range)
     import std.range;
     static struct S{int* p;}
     auto a = array(immutable(S).init.repeat(5));
+    assert(a.length);
```
Comment (JackStouffer) [inline std/array.d]: a.length == 5

File: std/array.d [hunk 3]
```diff
@@ -1209,7 +1215,7 @@ private template isInputRangeOrConvertible(E)
             insertInPlace(arr, 1, Int(2), Int(3));
             return equal(arr, [1, 2, 3, 4, 5]);  //check it works with postblit
         }
-        enum E = testctfe();
+        assert(testctfe());
```
Comment (JackStouffer) [inline std/array.d]: why not just delete this whole version none block?

File: std/array.d (hunk 1 above)
Comment (ZombineDev) [inline std/array.d]: In almost every C-like language?

Comment (RazvanN7) [inline std/array.d]: If you do not use the parameter, yes. It is especially useful when using design patterns (for example the visitor pattern).

File: std/array.d (hunk 3 above)
Comment (RazvanN7) [inline std/array.d]: I thought that it's there for the sake of illustration. I will delete it.

File: std/array.d (hunk 2 above)
Comment (RazvanN7) [inline std/array.d]: oh, missed it.

File: std/array.d [hunk 4]
```diff
@@ -3661,7 +3672,7 @@ unittest
     {
         return [new shared A].inputRangeObject;
     }
-    auto res = foo.array;
+    static assert(__traits(compiles, foo.array));
```
Comment (JackStouffer) [inline std/array.d]: Why are you removing traits compiles in some areas but adding them in others?

shouldn't this be `assert(foo.array == [new shared A]);`?

Comment (RazvanN7) [inline std/array.d]: I didn't remove any traits compiles in this PR. The line I deleted here was declaring res just to make sure that foo.array compiles. Nesting it with a traits compiles gets rid of the unused variable while preserving functionality

Comment (JackStouffer) [inline std/array.d]: >I didn't remove any traits compiles in this PR

https://github.com/dlang/phobos/pull/5676/files#diff-d7e888ff77123ed1925838a861e48e84L1503

>Nesting it with a traits compiles gets rid of the unused variable while preserving functionality

Better to actually test the results than get rid of the variable.

Comment (wilzbach) [inline std/array.d]: Yes, please test the actual result instead of using `compiles`

**Comment by RazvanN7** [comment]
@wilzbach Done, thank you.

**Pull request merged**